### ISSUE TITLE
Focus redirection does not work for mat-radio-button

### ISSUE THREAD

**Issue by jelbourn** [angular/material2#13953]
[mat-radio-button has some code](https://github.com/angular/material2/blob/5c0a061b98ca46eadcc066b0d926a972bd3fcce4/src/lib/radio/radio.ts#L341-L344) to redirect focus to the underlying native `<input type="radio">`. This seems to not actually work, though, because calling `.focus()` on the `<mat-radio-button>` does not trigger a `focus` event. 

[You can see the behavior here, with radio buttons inside of a dialog with cdkFocusInitial](https://stackblitz.com/edit/angular-fpxrwh-bzagqw?file=app/dialog-overview-example-dialog.html)

The same problem almost certainly exists for `mat-checkbox` and `mat-button-toggle`.

I'm not sure the best way to approach this, but one idea would be to use `focusin` or `useCapture` on a top-level element and specifically check for the controls in question.
